docs/Deprecating-Disabling-and-Removing-Formulae.md

```diff
@@ -0,0 +1,110 @@
+# Deprecating, Disabling, and Removing Formulae
+
+There are many reasons why formulae may be deprecated, disabled, or removed. This document explains the differences between each method as well as explaining when one method should be used over another.
+
+## Overview
+
+This general rule of thumb can be followed:
+
+- `deprecate!` should be used for formulae that _should_ no longer be used
+- `disable!` should be used for formulae that _cannot_ be used
+- Formulae that have been disabled for a long time or have a more serious issue should be removed
+
+## Deprecation
+
+If a user attempts to install a deprecated formula, they will be shown a warning message but the install will succeed.
+
+A formula should be deprecated to indicate to users that the formula should not be used and may be disabled in the future. Deprecated formulae should still be able to build from source and all bottles should continue to work. Users who choose to install deprecated formulae should not have any issues.
+
+The most common reasons for deprecation are when the upstream project is deprecated, unmaintained, or archived.
+
+To deprecate a formula, add a `deprecate!` call. This call should include a deprecation date (in the ISO 8601 format) and a deprecation reason:
+
+```ruby
+deprecate! date: "YYYY-MM-DD", because: :reason
+```
+
+The `date` parameter should be set to the date that the project became (or will become) deprecated. If there is no clear date but the formula needs to be deprecated, use today's date. If the `date` parameter is set to a date in the future, the formula will not become deprecated until that date. This can be useful if the upstream developers have indicated a date where the project will stop being supported.
```

Comment:
Versioned formula projects are still going, it's just the old version that got deprecated.

```suggestion
The `date` parameter should be set to the date that the formula source became (or will become) deprecated. If there is no clear date but the formula needs to be deprecated, use today's date. If the `date` parameter is set to a date in the future, the formula will not become deprecated until that date. This can be useful if the upstream developers have indicated a date where the project will stop being supported.
```